packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -205,7 +205,7 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate,
 
       let layoutNode = this.buildChild(node, 0, y);
       y = layoutNode.layoutInfo.rect.maxY;
-      nodes.push(layoutNode);
+      this.validRect.height && nodes.push(layoutNode);
```

Comment:
what is the height at this point? 0? undefined?
I think it'd be better to be a little more explicit and check if it's `> 0`, and if it's undefined, it'd be good to add a `!isNaN`